scripts/add-ocp-labels/main.go

```diff
@@ -0,0 +1,634 @@
+package main
+
+import (
+	goerror "errors"
+	"fmt"
+	"github.com/blang/semver"
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"text/template"
+
+	apimanifests "github.com/operator-framework/api/pkg/manifests"
+	apivalidation "github.com/operator-framework/api/pkg/validation"
+	"github.com/operator-framework/api/pkg/validation/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const ocpLabelAnnotation = "com.redhat.openshift.versions"
+
+// OCP version where the apis v1beta1 is no longer supported
+const ocpVerV1beta1Unsupported = "4.9"
+
+type BundleAnnotations struct {
+	Annotations map[string]string `yaml:"annotations,omitempty"`
+}
+
+type UpdateGraph struct {
+	UpdateGraph string `yaml:"annotations,omitempty"`
+}
+
+type BundleDir struct {
+	Path  string
+	Error string
+}
+
+type Output struct {
+	Errors                       []BundleDir
+	RequiresUpdate               []BundleDir
+	RequiresUpdatePkgFormat      []BundleDir
+	BundleUpdated                []BundleDir
+	BundleUnload                 []BundleDir
+	BundlesConfiguredAccordingly []BundleDir
+	BundlesMigrated              []BundleDir
+	UpdatedUsingServerMode       []BundleDir
+	PkgManifestUsingServerMode   []BundleDir
+	CurrentPath                  string
+}
+
+func main() {
+	createOutput()
+}
+
+func createOutput() {
```

Comment:
nit: I guess it depends on whether you see this code being used outside this scope. But I would suggest:
1. moving the currentPath stuff to the main
2. making createOutput a simple public constructor/factory function that takes the currentPath as a parameter
3. moving everything from 62 to 71 to some internal, but public, function, e.g. ProcessBundles or ApplyOCPLabel or something like that. You could also consider splitting the processing from the report preparation/writing.
4. put it all together in main